dali/pipeline/workspace/host_workspace.h

```diff
@@ -24,14 +24,15 @@
 #include "dali/pipeline/data/backend.h"
 #include "dali/pipeline/data/tensor.h"
 #include "dali/pipeline/util/thread_pool.h"
+#include "dali/pipeline/data/tensor_vector.h"
 #include "dali/pipeline/workspace/workspace.h"
 
 namespace dali {
 
 template <typename Backend>
-using HostInputType = vector<shared_ptr<Tensor<Backend>>>;
+using HostInputType = shared_ptr<TensorVector<Backend>>;
```

Comment:
It can be unique_ptr or a static allocation.

Reply:
All buffers are passed around as shared_ptrs, having this as outermost shared_ptr unifies the situation.